<!-- Launch the nodes and populate the ROS params needed for Spot to navigate -->
<launch>

    <!-- Require users to manually specify where to find the list of waypoints -->
    <arg name="waypoints_yaml" />

    <!-- Indicate that the navigation system is active -->
    <param name="/spot/navigation/active" value="true" />

    <!-- YAML file specifying the 2D poses of named waypoints -->
    <param name="/spot/navigation/waypoints_yaml" value="$(arg waypoints_yaml)" />

    <!-- Distance (meters) within which Spot is considered to have reached its navigation goal -->
    <param name="/spot/navigation/reached_goal_m" value="0.3" />

    <!-- Absolute angle (radians) within which Spot is considered to have reached its goal pose -->
    <param name="/spot/navigation/reached_goal_rad" value="0.2" />

    <!-- Duration (seconds) after which an entire navigation command will time out -->
    <param name="/spot/navigation/timeout_s" value="60" />

    <!-- Duration (seconds) after which the final local adjustment will time out -->
    <param name="/spot/navigation/adjustment_timeout_s" value="10" />

    <!-- Launch Spot's navigation stack using move_base -->
    <include file="$(find spot_move_base)/launch/move_base.launch"/>

</launch>